src/goto-programs/goto_sideeffects.cpp

```diff
@@ -726,9 +726,9 @@ void goto_convertt::remove_cpp_new(
 
   symbolt new_symbol;
 
-  new_symbol.name = "new_ptr$" + std::to_string(++temporary_counter);
+  new_symbol.name = "new_ptr$" + std::to_string(++tmp_symbol.counter);
```

Comment:
do we need to manually increment the counter here? Shouldn't we be calling new_symbol?

Reply:
See above.